ngrok/src/online_tests.rs

```diff
@@ -731,6 +731,60 @@ async fn verify_upstream_tls() -> Result<(), Error> {
     Ok(())
 }
 
+#[cfg_attr(not(feature = "online-tests"), ignore)]
+#[test]
+async fn session_root_cas() -> Result<(), Error> {
+    // host cannot validate cert
+    let resp = Session::builder()
+        .authtoken_from_env()
+        .root_cas("host")?
+        .connect()
+        .await;
+    assert!(resp.is_err());
+    let err_str = resp.err().unwrap().to_string();
+    tracing::debug!(?err_str);
+    assert!(err_str.contains("tls")); // tls issue
+
+    // default of 'trusted' cannot validate the marketing site
+    let resp = Session::builder()
+        .authtoken_from_env()
+        .server_addr("ngrok.com:443")?
+        .connect()
+        .await;
+    assert!(resp.is_err());
+    let err_str = resp.err().unwrap().to_string();
+    tracing::debug!(?err_str);
+    assert!(err_str.contains("tls")); // tls issue
+
+    // host can validate the marketing site
```

Comment:
should this be `cannot` validate?

Reply:
For this case `host` can validate, since it is pointing at `ngrok.com`, and it's asserting that the error that happens does not have `tls` in it. There's still an error since it's hitting the website instead of `tunnel.v2`, but it gets passed tls validation so the test is happy.

Reply:
I've updated the comments to hopefully be more clear